ios/FluentUI/Vnext/Button/Button.swift

```diff
@@ -126,55 +126,120 @@ struct FluentButtonBody: View {
+            backgroundColor = tokens.disabledBackgroundColor
+        } else if isPressed {
+            iconColor = tokens.highlightedIconColor
+            titleColor = tokens.highlightedTitleColor
+            borderColor = tokens.highlightedBorderColor
+            backgroundColor = tokens.highlightedBackgroundColor
+        } else {
+            iconColor = tokens.iconColor
+            titleColor = tokens.titleColor
+            borderColor = tokens.borderColor
+            backgroundColor = tokens.backgroundColor
+        }
 
-        return HStack(spacing: tokens.interspace) {
-            if let image = state.image {
-                Image(uiImage: image)
-                    .resizable()
-                    .foregroundColor(Color(isDisabled ? tokens.disabledIconColor :
-                                            (isPressed ? tokens.highlightedIconColor : tokens.iconColor)))
-                    .frame(width: tokens.iconSize, height: tokens.iconSize, alignment: .center)
+        @ViewBuilder
+        var buttonContent: some View {
+            HStack(spacing: tokens.interspace) {
+                if let image = state.image {
+                    Image(uiImage: image)
+                        .resizable()
+                        .foregroundColor(Color(iconColor))
+                        .frame(width: tokens.iconSize, height: tokens.iconSize, alignment: .center)
                 }
-
-            if let text = state.text {
-                Text(text)
-                    .fontWeight(.medium)
-                    .multilineTextAlignment(.center)
-                    .scalableFont(font: tokens.textFont,
-                                  shouldScale: !isFloatingStyle)
-                    .modifyIf(isFloatingStyle, { view in
-                        view.frame(minHeight: tokens.textMinimumHeight)
-                    })
+                if let text = state.text {
+                    Text(text)
+                        .fontWeight(.medium)
+                        .multilineTextAlignment(.center)
+                        .scalableFont(font: tokens.textFont,
+                                      shouldScale: !isFloatingStyle)
+                        .modifyIf(isFloatingStyle, { view in
+                            view.frame(minHeight: tokens.textMinimumHeight)
+                        })
                 }
+            }
+            .padding(tokens.padding)
+            .modifyIf(isFloatingStyle && !(state.text?.isEmpty ?? true), { view in
+                view.padding(.horizontal, tokens.textAdditionalHorizontalPadding )
+            })
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .foregroundColor(Color(titleColor))
+        }
+
+        @ViewBuilder
+        var buttonBackground: some View {
+            if tokens.borderSize > 0 {
+                buttonContent.background(
+                    AnyView(RoundedRectangle(cornerRadius: tokens.borderRadius)
```

Comment:
You don't need the expensive AnyView() wrapping anymore on the RoundedRectangles in the if/else cases.